petaio.c

```diff
@@ -745,19 +785,19 @@ static void register_io_blocks() {
 
     /* SF */
 #ifdef SFR
+    IO_REG_TYPE(Generation,       "u1", 1, 0);
+    IO_REG_TYPE(Generation,       "u1", 1, 4);
     IO_REG_WRONLY(StarFormationRate, "f4", 1, 0);
-#ifdef WINDS
-    IO_REG(StarFormationTime, "f4", 1, 4);
-#endif
-#ifdef METALS
-    IO_REG(Metallicity,       "f4", 1, 0);
-    IO_REG(Metallicity,       "f4", 1, 4);
-#endif /* METALS */
+    IO_REG_NONFATAL(BirthDensity, "f4", 1, 4);
```

Comment:
The handling of fatal can be improved. I think we need some overall flag to store if any block is successfully read. This shall be set during read_internal(). Then init.c can make use of that information to either die or generate some fake data to make it up. 

Also, when a block is not required, but data is provided, and corruption is detected during IO, we shall die instead of proceeding as if the block is not supplied.

Reply:
How do we detect corruption? Through the bigfile error code?

Reply:
Yes. If the file opens (exists) but the read fails, then it is a corruption.

Reply:
Not so - the open succeeds even if no file exists. It just creates an empty directory and it is the read that returns an error.

Reply:
That's a bigfile bug. I don't think it is supposed to ever do that...

Reply:
big_block_open will try to open 'header' file -- and it if doesn't exist it should throw an error. (line 315 of bigfile.c.)  big_file_open do work even if there is no dataset inside the file, but it is not the function you are calling here..

Reply:
I checked and indeed you are right  - I think I must have had a left-over folder from some earlier crashing run before. I will fix the error case.

Reply:
This is the only remaining part of this PR.

Reply:
Current head of this branch only returns if the open fails, and does endrun if read or seek fail. So I think it is good.